hyperspy/_signals/hologram_image.py

```
@@ -0,0 +1,329 @@
+        log_parameters : boolean
+            Logs reconstruction parameters
+
+        Returns
+        -------
+        wave : :class:`~hyperspy.signals.WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+
+        Notes
+        -----
+
+        """
+
+        # Parsing reference:
+        if not isinstance(reference, HologramImage):
+            if isinstance(reference, Signal2D):
+                _logger.warning('The reference image signal type is not HologramImage. It will '
+                                'be converted to HologramImage automatically.')
+                reference.set_signal_type('hologram')
+            elif reference is not None:
+                reference = HologramImage(reference)
+
+        # Testing match of navigation axes of reference and self (exception: reference nav_dim=1):
+        if (reference and not reference.axes_manager.navigation_shape == self.axes_manager.navigation_shape
+            and reference.axes_manager.navigation_size):
+
+            raise ValueError('The navigation dimensions of object and reference holograms do not match')
+
+        # Parsing sideband position:
+        if sb_position is None:
+            warnings.warn('Sideband position is not specified. The sideband will be found automatically which may '
+                          'cause wrong results.')
+            if reference is None:
+                sb_position = self.find_sideband_position(sb=sb)
+            else:
+                sb_position = reference.find_sideband_position(sb=sb)
+
+        else:
+            if not isinstance(sb_position, Signal1D):
+                sb_position = Signal1D(sb_position)
+
+        if sb_position.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_position.axes_manager.navigation_size:
+                warnings.warn('Sideband position dimensions do not match neither reference nor hologram dimensions.'
```

I would rather raise an error in this case as the input is incorrect.